### PR TITLE
Add a tool to consistently format embargoed patches

### LINKED ISSUE #189: Create a script to consistently format patch files
Based on feedback from distributors, we should automate creating consistent patch files when sending embargoed patches to distributors-announce@

* The patches should be a single commit (where possible)
* The patches should have any revision added to the file name (e.g. if we need to send out an updated patch, it should have a different filename than the original)
* The patches should be in a .zip file when sent via email
* The metadata at the top of the .patch file should look like:
```
From: distributors-announce <distributors-announce@kubernetes.io>
Date: Tue, 31 Oct 2023 18:23:55 +0000
Subject: Fix CVE-20XX-XXXX
​
---
```

### PR DESCRIPTION
**Pull request by cji**
Fixes #189 

I tested this with the most recent round of embargoed patches, but would encourage reviewers to test for others as well if you have the time please.

Feedback welcome on making this seem like less of a hack, but this was a quick way I got it worked as expected.

### CODE REVIEW

File: tools/patchformat.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+parser = argparse.ArgumentParser(description='Format patch files')
+required = parser.add_argument_group('required arguments')
+required.add_argument('--pr', help='Pull Request Number in kubernetes-security/kubernetes with the fix (e.g. 63)', required=True)
+required.add_argument('--cve', help='CVE for the vulnerability', required=True)
+required.add_argument('-r', '--release', help='Kubernetes release branch (e.g. 1.30)', required=True)
+parser.add_argument('-v', '--version', help='Patch version, if a correction is being sent')
+args = parser.parse_args()
+
+pull = api.pulls.get(args.pr)
```
Comment (ritazh) [inline tools/patchformat.py:19]: I'm getting 404 with my token. I wonder if I'm missing certain permissions off the repo? Lets pair on this when you have time.

Comment (cji) [reply tools/patchformat.py:19]: I checked and you have admin on the repo but maybe the token doesn't have permission? I'm back from vacation this week and can pair when you're free!

File: tools/patchformat.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import argparse
+from ghapi.core import GhApi
+import os
+import time
+
+username = os.environ['GH_USERNAME']
+github_token = os.environ['GH_TOKEN']
```
Comment (ritazh) [inline tools/patchformat.py:7]:
```suggestion
# When creating github personal access token, be sure to check all permissions under the repo scope to access private repos
github_token = os.environ['GH_TOKEN']
```

**Comment by ritazh** [comment]
One nit to add a comment, otherwise LGTM

**Review by ritazh — APPROVED** [review]
LGTM

**Pull request merged**